supernode/daemon/mgr/preheat_mgr.go

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright The Dragonfly Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mgr
+
+import (
+	"context"
+
+	"github.com/dragonflyoss/Dragonfly/apis/types"
+)
+
+// PreheatStatus
+type PreheatStatus string
+
+const (
+	// Waiting the preheat task is waiting for executing
+	Waiting PreheatStatus = "WAITING"
+
+	// Running the preheat task is running
+	Running PreheatStatus = "RUNNING"
+
+	// Success the preheat task is finished and successfully
+	Success PreheatStatus = "SUCCESS"
+
+	// Failed the preheat task is finished and failed
+	Failed PreheatStatus = "FAILED"
+)
+
+var preheatStatusEnums = []PreheatStatus{Waiting, Running, Success, Failed}
+
+// IsValid tests the value of the instance whether is valid.
+func (ps PreheatStatus) IsValid() bool {
+	for _, v := range preheatStatusEnums {
+		if v == ps {
+			return true
+		}
+	}
+	return false
+}
+
+// String return the string value of the instance.
+func (ps PreheatStatus) String() string {
+	return (string)(ps)
+}
+
+// PreheatTask store the detailed preheat task information.
+type PreheatTask struct {
+	ID         string
+	URL        string
+	Type       string
+	Filter     string
+	Identifier string
+	Headers    map[string]string
+
+	ParentID   string
+	Children   []string
+	Status     PreheatStatus
+	StartTime  int64
+	FinishTime int64
+	ErrorMsg   string
+}
+
+// PreheatManager
+type PreheatManager interface {
+	// Create creates a preheat task to cache data in supernode, thus accelerating the
+	// p2p downloading.
+	Create(ctx context.Context, task *types.PreheatCreateRequest) (id string, err error)
+
+	// Get gets detailed preheat task information by id.
+	Get(ctx context.Context, id string) (task *PreheatTask, err error)
+
+	// Delete deletes a preheat task by id.
+	Delete(ctx context.Context, id string) (err error)
```

Comment:
Do we need to add the `Delete` endpoint if the swagger.yml? Since I have not found that. 
I am afraid that we should define the API in swagger.yml first, and then implement that in code.

Reply:
done